connector/spanmetricsconnector/factory.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanmetricsconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector"
+
+import (
+	"context"
+	"time"
+
+	"github.com/tilinna/clock"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "spanmetrics"
+	// The stability level of the connector.
+	stability = component.StabilityLevelDevelopment
+)
+
+// NewFactory creates a factory for the spanmetrics connector.
+func NewFactory() connector.Factory {
+	return connector.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		connector.WithTracesToMetrics(createConnector, stability),
```

Comment:
This component emits both traces and metrics. I think we can go for `connector.WithTracesToMetrics` and `connector.WithTracesToTraces`. I am not sure, but it should be able to drop this lock [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18381/files#diff-4cf1d004574eff4ea9a26ff033740136245993c942742b4e8c6622e522599910R242-R258) after branching the pipeline.